main.rb

```diff
@@ -0,0 +1,168 @@
+# Imports
+require './book'
+require './student'
+require './teacher'
+require './rental'
+
+# Welcome note
+puts "---------------------------------\n
+ Welcome to OOP School Library App!\n
+ ------------------------------"
+
+# State Helpers
+@book_arr = []
+@person_arr = []
+
+# Method to add new book record
+def add_book(book)
+  @book_arr.push(book)
+  @book_arr
+end
+
+# Method to add new book record
+def add_person(person)
+  @person_arr.push(person)
+  @person_arr
+end
+
+# Method to create new student's record
+def create_student
+  print 'Age: '
+  age = gets.chomp
+  print 'Name: '
+  name = gets.chomp
+  print 'Has parent permission? [Y/N]: '
+  perm_input = gets.chomp
+  case perm_input.upcase
+  when 'Y'
+    permission = true
+  when 'N'
+    permission = false
+  else
+    puts 'Invalid entry!'
+    init
+  end
+  student = Student.new(age, name)
+  student.parent_permission = permission
+  student.type = 'Student'
+  add_person(student)
+  success('Person')
+end
+
+# Method to create new teacher record
+def create_teacher
+  print 'Age: '
+  age = gets.chomp
+  print 'Name: '
+  name = gets.chomp
+  teacher = Teacher.new(age, name)
+  print 'Specialization: '
+  specialization = gets.chomp
+  teacher.specialization = specialization
+  teacher.type = 'Teacher'
+  add_person(teacher)
+  success('Person')
+end
+
+# Confirmation notes
+def success(item)
+  puts "#{item} created succesfully!"
+end
+
+# Menu Operations
+def list_all_books
+  @book_arr.map { |book| puts "Title: \"#{book.title}\" Author: #{book.author}" }
+  init
+end
+
+# Method to list all persons' record
+def list_all_people
+  @person_arr.map { |person| puts "[#{person.type}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+  init
+end
+
+# Method to create new person
+def create_a_person()
+  puts 'Do you want to create a student (1) or a teacher (2)? [enter a number]: '
+  option = gets.chomp
+  case option.to_i
+  when 1
+    create_student
+  when 2
+    create_teacher
+  else
+    puts 'Invalid entry !'
+  end
+  init
+end
+
+# Method to create a new book
+def create_a_book
+  print 'Title: '
+  title = gets.chomp
+  print 'Author: '
+  author = gets.chomp
+  book = Book.new(title, author)
+  add_book(book)
+  success('Book')
+  init
+end
+
+# Method to create a new rental
+def create_a_rental
+  puts 'Select a book from the following list by number'
+  @book_arr.each_with_index { |book, idx| puts "#{idx}) Title: \"#{book.title}\" Author: #{book.author}" }
+  book_index = gets.chomp.to_i
+  puts 'Select a person from the following list by number (not id)'
+  @person_arr.each_with_index do |person, idx|
+    puts "#{idx}) [#{person.type}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+  end
+  person_index = gets.chomp.to_i
+  print 'Date (yyyy/mm/dd): '
+  date = gets.chomp
+  person = @person_arr[person_index]
+  book = @book_arr[book_index]
+  Rental.new(date, person, book)
+  success('Rental')
+  init
+end
+
```

Comment:
- [ ] I had an issue too while creating the rentals.

![image](https://user-images.githubusercontent.com/55635977/158031080-fde9ceca-0b3c-4066-839c-6f17655b5897.png)

Reply:
You have to create person and book for creating new rentals